features/consent/src/main/java/in/projecteka/jataayu/consent/ui/fragment/RequestedConsentDetailsFragment.kt

```diff
@@ -232,4 +241,36 @@ class RequestedConsentDetailsFragment : BaseFragment(), ItemClickCallback,
             }
         }
     }
+
+    fun validateUser() {
+        if (context?.getPinCreated()!!) {
+            val intent = Intent(context, PinVerificationActivity::class.java)
+            startActivityForResult(intent, 301)
+        } else {
+            val intent = Intent(context, CreatePinActivity::class.java)
+            startActivityForResult(intent, 201)
+        }
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (resultCode == Activity.RESULT_OK){
+            if (requestCode == 201) {
+                activity?.setPinCreated(true)
+                grantConsent()
+            }  else if (requestCode == 301){
+                grantConsent()
+            }
+        }
+    }
+
+    private fun getNameOf(hiu: HipHiuIdentifiable, completion: () -> Unit) {
```

Comment:
HipHiuIdentifiable is an Interface and it has ID to identify HIP and HIU.  HIP, HIU, HIURequestor extends this interface. Function params have HipHiuIdentifiable rather than string. so you can avoid passing wrong identifier or some random string.